File: frontend/src/service/api/permission.ts
import { request } from '../request';

export interface PermissionItem {
  code: string;
  systemCode: 'store' | 'rag' | 'system';
  moduleName: string;
  operation: string;
  description: string;
}

export interface BusinessRoleItem {
  id: number;
  code: string;
  name: string;
  active: boolean;
  systemRole: boolean;
  permissionCodes: string[];
}

export interface PermissionUserItem {
  id: number;
  username: string;
  role: 'USER' | 'ADMIN';
}

export interface UserAuthorization {
  userId: number;
  roleIds: number[];
  overrides: Array<{ permissionCode: string; effect: 'GRANT' | 'DENY' }>;
  effectivePermissions: string[];
  orgTags: string;
  primaryOrg: string;
}

export function fetchPermissionCatalog() {
  return request<PermissionItem[]>({ url: '/admin/permissions/catalog' });
}

export function fetchBusinessRoles() {
  return request<BusinessRoleItem[]>({ url: '/admin/permissions/roles' });
}

export function fetchCreateBusinessRole(data: { code: string; name: string; permissionCodes: string[] }) {
  return request({ url: '/admin/permissions/roles', method: 'post', data });
}

export function fetchUpdateBusinessRolePermissions(roleId: number, permissionCodes: string[]) {
  return request({ url: `/admin/permissions/roles/${roleId}/permissions`, method: 'put', data: { permissionCodes } });
}

export function fetchPermissionUsers() {
  return request<PermissionUserItem[]>({ url: '/admin/users' });
}

export function fetchUserAuthorization(userId: number) {
  return request<UserAuthorization>({ url: `/admin/permissions/users/${userId}` });
}

export function fetchUpdateUserAuthorization(
  userId: number,
  data: { roleIds: number[]; grants: string[]; denies: string[]; orgTags: string[]; primaryOrg: string }
) {
  return request<UserAuthorization>({ url: `/admin/permissions/users/${userId}`, method: 'put', data });
}

File: src/main/java/com/yizhaoqi/smartpai/model/Permission.java
package com.yizhaoqi.smartpai.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "permissions")
/** 定义跨眼镜店、RAG 和系统管理使用的稳定功能权限目录。 */
public class Permission {

    @Id
    @Column(length = 120)
    private String code;

    @Column(nullable = false, length = 40)
    private String systemCode;

    @Column(nullable = false, length = 60)
    private String moduleName;

    @Column(nullable = false, length = 40)
    private String operation;

    @Column(nullable = false, length = 200)
    private String description;
}
